3.5-windowsservercore-1709/runtime/Dockerfile

```diff
@@ -1,11 +1,37 @@
+# escape=`
+
 FROM microsoft/windowsservercore:1709
 
-ADD install /build
-RUN DISM /Online /Add-Package /PackagePath:C:\build\microsoft-windows-netfx3-ondemand-package.cab & \
-    DISM /Online /Add-Package /PackagePath:C:\build\patch\Windows10.0-KB3213986-x64.cab  & \
-    rd /s /q C:\build
-RUN set COMPLUS_NGenProtectedProcess_FeatureEnabled=0 & \
-    \Windows\Microsoft.NET\Framework64\v4.0.30319\ngen update & \
-    \Windows\Microsoft.NET\Framework\v4.0.30319\ngen update & \
-    \Windows\Microsoft.NET\Framework64\v2.0.50727\ngen update & \
+# Install .NET Fx 3.5
+RUN powershell -Command `
+        $ErrorActionPreference = 'Stop'; `
+        $ProgressPreference = 'SilentlyContinue'; `
+        Invoke-WebRequest `
+            -UseBasicParsing `
+            -Uri "https://dotnetbinaries.blob.core.windows.net/dockerassets/microsoft-windows-netfx3-1709.zip" `
+            -OutFile microsoft-windows-netfx3.zip; `
+        Expand-Archive microsoft-windows-netfx3.zip; `
+        Remove-Item -Force microsoft-windows-netfx3.zip; `
+        Add-WindowsPackage -Online -PackagePath .\microsoft-windows-netfx3\microsoft-windows-netfx3-ondemand-package.cab; `
```

Comment:
Add the `-NoRestart` flag

Reply:
I’m not certain we should be specify the /NoRestart flag.  Whatever we install should be finalized.  You don’t want any installation activities to happen each time the container is started (I’m not certain this is even possible).  By not specifying /NoRestart, the Docker build will hang/fail if/when a restart is required, thus we will know something is "wrong".  When /NoRestart is specified, the build completes and a warning is emitted stating there is a pending restart.